### DISCUSSION TITLE
difference "PUBLIC_INSTANCE_URL" & "PUBLIC_INTERNAL_INSTANCE"

### DISCUSSION THREAD

**Discussion by Tealk** [Xyphyn/photon#315] [Q&A] [answered]
Hi,

I would like to know what the difference is between the following two variables:
- `PUBLIC_INSTANCE_URL`
- `PUBLIC_INTERNAL_INSTANCE`

**Comment by HEPOSHEIKKI** [comment]
https://github.com/Xyphyn/photon/blob/e9ee6186428912d0223b86316ffb049c45e61b45/README.md?plain=1#L51

The PUBLIC_INTERNAL_INSTANCE variable defines where the Photon instance is hosted, and is used when the Lemmy instance is at a different address. It's used for links in Photon, so for example when clicking a post you will be sent to the URL `photon.lemmy.world` instead of `lemmy.world`. It is not required when the Photon instance is hosted on the same URL as the Lemmy instance.

**Reply by Tealk** [reply]
And in which scenario would the url for `PUBLIC_INSTANCE_URL` `PUBLIC_INTERNAL_INSTANCE` be different?

**Comment by HEPOSHEIKKI** [comment]
> And in which scenario would the url for `PUBLIC_INSTANCE_URL` `PUBLIC_INTERNAL_INSTANCE` be different?
For example the photon instance on lemmy.world

The lemmy instance is at lemmy.world and the photon instance is at photon.lemmy.world

**Reply by Tealk** [reply]
I did that too, but only with PUBLIC_INSTANCE_URL = https://photon.rollenspiel.forum

- https://rollenspiel.forum
- https://photon.rollenspiel.forum

**Comment by Xyphyn** [comment, marked as answer]
When SSR is enabled, that is the URL requests are made to by the server. Without it, it will default to PUBLIC_INSTANCE_URL
